src/ray/raylet/scheduling_queue.cc

```diff
@@ -72,6 +72,42 @@ std::vector<Task> SchedulingQueue::RemoveTasks(std::unordered_set<TaskID> task_i
   return removed_tasks;
 }
 
+void SchedulingQueue::MoveTasks(std::unordered_set<TaskID> task_ids, TaskState src_state,
+                                TaskState dst_state) {
+  // TODO(atumanov): check the states first to ensure the move is transactional.
+  std::vector<Task> removed_tasks;
```

Comment:
the `removeTasksFromQueue` needs a container to place the removed tasks into. I don't think it can fail, it may just return an empty set, if no such tasks were found.